File: Day_2/day_2.R
library(readr)
library(stringr)
library(dplyr)

dat <- read_delim("Day_2/day_2_dat.txt", " ", col_names = FALSE)
# Part 1
passwords <- dat %>% 
  mutate(X2 = str_remove(X2, ":"),
         low = str_extract(X1, "^\\d{1,2}") %>% as.integer(),
         high = str_extract(X1, "\\d{1,2}$") %>% as.integer(),
         n = str_count(X3, X2)) %>%
  filter(n >= low, n <= high)
dim(passwords)[1]

# Part 2 
passwords <- dat %>% 
  mutate(X2 = str_remove(X2, ":"),
         loc1 = str_extract(X1, "^\\d{1,2}") %>% as.integer(),
         loc2 = str_extract(X1, "\\d{1,2}$") %>% as.integer(),
         test1 = str_sub(X3, start = loc1, end = loc1) == X2,
         test2 = str_sub(X3, start = loc2, end = loc2) == X2) %>% 
  filter(test1 + test2 == 1)
dim(passwords)[1]

File: Day_2/day_2.py
# Day 2
# Read in data 
dat = open("/Users/christina/Documents/adventofcode_2020/Day_2/day_2_dat.txt").readlines()

# Part 1
i = 0 
for x in dat: 
    nums, letter, password = x.split()
    nums = [int(x) for x in nums.split("-")]
    letter = letter[0]
    n = password.count(letter)
    if nums[0] <= n <= nums[1]:
        i = i + 1

print(i)   

# Part 2
i = 0 
y = 0 
for x in dat: 
    nums, letter, password = x.split()
    nums = [int(x)-1 for x in nums.split("-")]
    letter = letter[0]
    test1 = password[nums[0]] == letter
    test2 = password[nums[1]] == letter
    if (test1 or test2):
        i = i + 1
    if (test1 and test2):
        y = y + 1

print(i-y) 